internal/rbd/rbd_util.go

```diff
@@ -1918,3 +1924,18 @@ func genVolFromVolIDWithMigration(
 
 	return rv, err
 }
+
+// setVolumeMetadata set PV/PVC/PVCNamespace metadata on RBD image.
+func (rv *rbdVolume) setVolumeMetadata(parameters map[string]string) error {
+	keys := []string{pvcNameKey, pvcNamespaceKey, pvNameKey}
```

Comment:
Use 
```go
for k, v := range k8s.GetVolumeMetadata(parameters) {
    err := rv.SetMetadata(k, v)
}
```

That way the kubernetes specific metadata filtering is kept inside the k8s package.